dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java

```diff
@@ -398,13 +400,19 @@ private void processQuery(String filter, String sortCol, String sortOrder, Crite
                     query.orderBy(cb.asc(cb.size(entry.<Collection>get("starredUsers"))), cb.desc(entry.get("id")));
                 }
             } else {
-                Path<Object> sortPath = entry.get(sortCol);
-                if (!Strings.isNullOrEmpty(sortOrder) && "desc".equalsIgnoreCase(sortOrder)) {
-                    query.orderBy(cb.desc(sortPath), cb.desc(entry.get("id")));
-                    predicates.add(sortPath.isNotNull());
-                } else {
-                    query.orderBy(cb.asc(sortPath), cb.desc(entry.get("id")));
-                    predicates.add(sortPath.isNotNull());
+                try {
+                    Path<Object> sortPath = entry.get(sortCol);
+                    if (!Strings.isNullOrEmpty(sortOrder) && "desc".equalsIgnoreCase(sortOrder)) {
+                        query.orderBy(cb.desc(sortPath), cb.desc(entry.get("id")));
+                        predicates.add(sortPath.isNotNull());
+                    } else {
+                        query.orderBy(cb.asc(sortPath), cb.desc(entry.get("id")));
+                        predicates.add(sortPath.isNotNull());
+                    }
+                } catch (IllegalArgumentException e) {
+                    LOG.warn("Could not get published entries due to an invalid sortCol value. Error is ", e);
+                    throw new CustomWebApplicationException("Could not get published entries due to an invalid sortCol value. Error is "
```

Comment:
1. You could create the same message once for this line and the next line (and they're slightly different now).
2. This method is called places, although they currently do end up being called from the same endpoint. Still, I feel the `published entries` is a little too specific, e.g., if we ever invoke this from somewhere else.

Reply:
Changed! Now its `Could not process query due to the invalid sortCol value.` Thanks!